### PR TITLE
Ports tg's del the world unit test

### PR DESCRIPTION
**Pull request by ivanmixo**
## About The Pull Request

Requires https://github.com/tgstation/TerraGov-Marine-Corps/pull/13824

https://github.com/tgstation/tgstation/pull/57728/files
https://github.com/tgstation/tgstation/pull/58972/files
https://github.com/tgstation/tgstation/pull/59612/files <- main one
https://github.com/tgstation/tgstation/pull/62725/files
https://github.com/tgstation/tgstation/pull/63877/files
https://github.com/tgstation/tgstation/pull/69696/files <- small bit, this one should be ported fully but is out of scope
https://github.com/tgstation/tgstation/pull/74604/files
https://github.com/tgstation/tgstation/pull/75535/files
https://github.com/tgstation/tgstation/pull/77504/files
https://github.com/tgstation/tgstation/pull/76214/files
https://github.com/tgstation/tgstation/pull/70729/files
https://github.com/tgstation/tgstation/pull/77850/files

\+ more, probably. I've marked the major ones. 

Huge unit test that spawns almost every item in the game and then deletes it. This catches a *lot* of errors and harddels.
## Why It's Good For The Game

More unit tests good.
## Changelog
:cl:
code: Ported tg's del the world unit test and brought our reference tracker and garbage collector mostly up to date. Also squashed a lot of harddels.
/:cl:

### CODE REVIEW

**Comment by ivanmixo** [comment]
This should seriously be TMd btw. Don't yolomerge. I just need to add some comments and update the PR desc

**Comment by TiviPlus** [comment]
I was about to click it lol

**Comment by ivanmixo** [comment]
back to hell

File: code/modules/organs/organ_internal.dm [hunk 1]
```diff
@@ -256,11 +256,14 @@
 
 /datum/internal_organ/kidneys/New(mob/living/carbon/carbon_mob)
 	. = ..()
+	if(!.)
+		return
```
[the remark this answers is not in the document]
Comment (ivanmixo) [reply code/modules/organs/organ_internal.dm]: No clue why I did it like this

File: code/modules/reagents/holder.dm [hunk 2]
```diff
@@ -811,3 +811,7 @@
 				random_reagents += R
 	var/picked_reagent = pick(random_reagents)
 	return picked_reagent
+
+/atom/movable/chem_holder
```
Comment (QualityVan) [inline code/modules/reagents/holder.dm:815]: why
why not just use a generic AM if obj doesn't suit for some reason?

Comment (ivanmixo) [reply code/modules/reagents/holder.dm:815]: idk why tg did it like this lol

**Review by Lumipharon — APPROVED** [review]
lets go

**Pull request merged**